### PR TITLE
Refresh POM and dependencies

### PR DESCRIPTION
**Pull request by jglick**
See https://github.com/jenkinsci/cloudbees-folder-plugin/pull/165#issuecomment-724716790. Supersedes #166, #167, and #168.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1 +1 @@
-buildPlugin(configurations: buildPlugin.recommendedConfigurations())
+buildPlugin(useAci: true)
```
Comment (fcojfernandez) [inline Jenkinsfile:1]: Why are you removing the `buildPlugin.recommendedConfigurations()`? I thought that was the correct way to execute the `buildPlugin` function

Comment (jglick) [reply Jenkinsfile:1]: https://github.com/jenkins-infra/pipeline-library/pull/145 and in particular it is impossible for someone without write access to the repo to accurately test changes to `jenkins.version` so long as the `Jenkinsfile` refers to other Jenkins versions.

**Review by timja — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (fcojfernandez) [reply Jenkinsfile:1]: Thanks for the explanation!

**Review by fcojfernandez — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -21,10 +21,10 @@
   <properties>
     <revision>6.15</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.176.1</jenkins.version>
+    <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
+    <jenkins.version>2.204.6</jenkins.version>
```
Comment (jglick) [inline pom.xml:25]: see https://github.com/jenkinsci/snakeyaml-api-plugin/pull/2#discussion_r473643918

**Pull request merged**